src/core/message.ts

```diff
@@ -115,22 +210,66 @@ export class DefaultMessage implements Message {
     public readonly createdAt: Date,
     public readonly metadata: MessageMetadata,
     public readonly headers: MessageHeaders,
+
+    public readonly latestAction: ChatMessageActions,
+    public readonly latestActionSerial: string,
+
+    public readonly deletedAt?: Date,
+    public readonly updatedAt?: Date,
+    public readonly latestActionDetails?: ActionDetails,
   ) {
-    this._calculatedTimeserial = DefaultTimeserial.calculateTimeserial(timeserial);
+    this._calculatedOriginTimeserial = DefaultTimeserial.calculateTimeserial(timeserial);
+    this._calculatedActionSerial = DefaultTimeserial.calculateTimeserial(latestActionSerial);
 
     // The object is frozen after constructing to enforce readonly at runtime too
     Object.freeze(this);
   }
 
+  get isUpdated(): boolean {
+    return this.updatedAt !== undefined;
+  }
+
+  get isDeleted(): boolean {
+    return this.deletedAt !== undefined;
+  }
+
+  get actionedBy(): string | undefined {
+    return this.latestActionDetails?.clientId;
```

Comment:
if action is create should this return the original author as a convenience?

Reply:
This has been removed anyway in favour of deletedBy and updatedBy.